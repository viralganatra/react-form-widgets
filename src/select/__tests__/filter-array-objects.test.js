import FilterArrayObjects from '../filter-array-objects';

const props = {
    items: [
        { label: 'Apple' },
        { label: 'Orange' },
        { label: 'Pear' },
        { label: 'Bear' },
    ],
    key: 'label',
}

describe('FilterArrayObjectsByKey', () => {
    it('should filter a list of items based on a key', () => {
        const filteredSingle = FilterArrayObjects({
            ...props,
            searchStr: 'apple',
        });

        const filteredMulti = FilterArrayObjects({
            ...props,
            searchStr: 'ear',
        });

        const filteredInvalidKey = FilterArrayObjects({
            ...props,
            key: 'foo',
            searchStr: 'ear',
        });

        expect(filteredSingle).toMatchSnapshot();
        expect(filteredMulti).toMatchSnapshot();
        expect(filteredInvalidKey).toMatchSnapshot();
    });

    it('should accept case sensitive values', () => {
        const filteredMatches = FilterArrayObjects({
            ...props,
            lowercase: false,
            searchStr: 'Apple',
        });

        const filteredNoMatches = FilterArrayObjects({
            ...props,
            lowercase: false,
            searchStr: 'apple',
        });

        expect(filteredMatches).toMatchSnapshot();
        expect(filteredNoMatches).toMatchSnapshot();
    });
});
