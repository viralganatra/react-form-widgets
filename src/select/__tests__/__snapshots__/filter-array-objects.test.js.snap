exports[`FilterArrayObjectsByKey should accept case sensitive values 1`] = `
Array [
  Object {
    "label": "Apple",
  },
]
`;

exports[`FilterArrayObjectsByKey should accept case sensitive values 2`] = `Array []`;

exports[`FilterArrayObjectsByKey should filter a list of items based on a key 1`] = `
Array [
  Object {
    "label": "Apple",
  },
]
`;

exports[`FilterArrayObjectsByKey should filter a list of items based on a key 2`] = `
Array [
  Object {
    "label": "Pear",
  },
  Object {
    "label": "Bear",
  },
]
`;

exports[`FilterArrayObjectsByKey should filter a list of items based on a key 3`] = `Array []`;
